openpgp/packet/signature.go

```diff
@@ -706,11 +764,13 @@ func (sig *Signature) Serialize(w io.Writer) (err error) {
 	length := len(sig.HashSuffix) - 6 /* trailer not included */ +
 		2 /* length of unhashed subpackets */ + unhashedSubpacketsLen +
 		2 /* hash tag */ + sigLength
+	if sig.Version() == 5 {
+		length -= 4 // eight-octet instead of four-octet big endian
+	}
```

Comment:
And then also here, I think this is not correct for binary and text signatures (the v5-specific fields are included in the length but shouldn't be). So instead of starting with the length of `sig.HashSuffix` and then subtracting stuff, I would compute the length from the ground up here (i.e. a constant plus the length of the hashed subpackets, plus the unhashed subpackets as it's already doing).

Reply:
This should work now, given that the length is consistent, it only takes into account the 4 octet v/s 8 octet difference.